hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaMessageReceiver.java

```diff
@@ -49,32 +45,23 @@ public KafkaMessageReceiver(String topicName, ConsumerConnector consumerConnecto
     }
 
     @Override
-    public RawMessage next() {
-        try {
+    public Message next() {
+        try (Timer.Context readingTimerContext = readingTimer.time()) {
             MessageAndMetadata<byte[], byte[]> message = iterator.next();
-            UnwrappedMessageContent unwrappedContent = contentWrapper.unwrapContent(message.message());
-            return new RawMessage(
-                    unwrappedContent.getStringFromMetadata(ID),
+            MessageWithMetadata unwrappedContent = contentWrapper.unwrapContent(message.message());
```

Comment:
I think `UnwrappedMessageContent` was a better name, now it could be confused with kafka's `MessageAndMetadata`